cmd/power/src/lib.rs

```diff
@@ -28,15 +35,84 @@ struct PowerArgs {
         parse(try_from_str = parse_int::parse)
     )]
     timeout: u32,
+
+    /// get phase current where available
+    #[clap(long)]
+    phase_current: bool,
 }
 
 struct Device<'a> {
     name: &'a str,
+    rail: Option<usize>,
     voltage: Option<usize>,
     current: Option<usize>,
     input_voltage: Option<usize>,
     input_current: Option<usize>,
     temperature: Option<usize>,
+    phases: Option<&'a Vec<u8>>,
+    phase_currents: Option<Vec<Option<f32>>>,
+}
+
+fn phase_currents(
+    hubris: &HubrisArchive,
+    core: &mut dyn Core,
+    context: &mut HiffyContext,
+    devices: &mut BTreeMap<(&String, HubrisSensorDevice), Device>,
+) -> Result<()> {
+    let op = hubris.get_idol_command("Power.phase_current")?;
+    let ok = hubris.lookup_basetype(op.ok)?;
+    let mut ops = vec![];
+
+    if ok.encoding != HubrisEncoding::Float {
```

Comment:
Nit: you could combine these conditionals, since I don't think we need to be _this_ fine-grained with error reporting
```suggestion
    if (ok.encoding, ok.size) != (HubrisEncoding::Float, 4) {
```

Reply:
Good call -- and also did this in the code that I copy-and-pasted it from, and then the places that I cut-and-pasted that code to. ;)